src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReplacementService.cs

```diff
@@ -366,7 +366,7 @@ private ReplacementSubjectMeta GetReplacementSubjectMeta(Guid subjectId)
                                         {
                                             Id = group.Key.Id,
                                             Label = group.Key.Label,
-                                            FilterItems = filter.ToList()
+                                            FilterItems = group.Key.FilterItems.Intersect(filter).ToList()
```

Comment:
Cool 👍 

Not sure if you followed why I changed this from `FilterItems = group.Key.FilterItems` to `FilterItems = filter.ToList()` in the first place but it was the bug in EES-1434 https://github.com/dfe-analytical-services/explore-education-statistics/pull/2024/files where I saw that the groups FilterItems was influenced by when that group was read. 

When fetched from a filterItem.Include(item => item.FilterGroup) for the first time then the group would only have the single item. If this or some other code had already fetched another item with the same group, or fetched the group itself and included all the items, then the groups items would include those unwanted ones too! We saw this when replacing multiple datablocks and footnotes at the same time with lots of criteria.

This change looks good and still filters out the unwanted ones :)